src/main/java/org/radarcns/management/service/MetaTokenService.java

```diff
@@ -103,20 +105,26 @@ public MetaToken getToken(String tokenName) throws CustomNotFoundException {
     }
 
     /**
-     * Builds a unique meta-token instance, by checking for token-name collision.
+     * Saves a unique meta-token instance, by checking for token-name collision.
+     * If a collision is detection, we try to save the token with a new tokenName
      * @return an unique token
      */
-    public MetaToken buildUniqueToken() {
-        MetaToken token = new MetaToken();
-
-        while (metaTokenRepository.findOneByTokenName(token.getTokenName()).isPresent()) {
-            token.tokenName(RandomStringUtils.randomAlphanumeric(SHORT_ID_LENGTH));
+    public MetaToken saveUniqueToken(String token, Boolean fetched, Instant expiryTime) {
+        MetaToken metaToken = new MetaToken()
```

Comment:
still need to add the `.tokenName(RandomStringUtils.randomAlphanumeric(SHORT_ID_LENGTH))` builder method

Reply:
The the MetaToken constructor assigns `RandomStringUtils.randomAlphanumeric(SHORT_ID_LENGTH)` to tokenName. That is why I didn't have it explicitly.